### PR TITLE
zephyr: allow dynamic numeration of flash_areas

### PR DESCRIPTION
**Pull request by nvlsianpu**
Zephyr flash_map reworks caused that areas id exact number are
assigned dynamically.

This patch i counterpart to
https://github.com/zephyrproject-rtos/zephyr/pull/8837
Merge after above PR.

Signed-off-by: Andrzej Puzdrowski <andrzej.puzdrowski@nordicsemi.no>

### CODE REVIEW

File: boot/zephyr/flash_map_extended.c [hunk 1]
```diff
@@ -48,12 +48,20 @@ int flash_device_base(uint8_t fd_id, uintptr_t *ret)
...
+ * MCUBoot uses continuous numbering for slot 0, slot 1, and the scratch
+ * while zephyr might numbering it differently.
  */
 int flash_area_id_from_image_slot(int slot)
 {
-    return slot + FLASH_AREA_IMAGE_0;
+    static const int area_id_tab[] = {FLASH_AREA_IMAGE_0, FLASH_AREA_IMAGE_1,
+                                      FLASH_AREA_IMAGE_SCRATCH};
+
+    if(slot >= 0 && slot < ARRAY_SIZE(area_id_tab)) {
```
Comment (utzig) [inline boot/zephyr/flash_map_extended.c]: missing space `if (`

File: boot/zephyr/flash_map_extended.c [hunk 2]
```diff
@@ -48,12 +48,20 @@ int flash_device_base(uint8_t fd_id, uintptr_t *ret)
 }
 
 /*
- * This depends on the mappings defined in sysflash.h, and assumes
- * that slot 0, slot 1, and the scratch areas are contiguous.
+ * This depends on the mappings defined in sysflash.h.
+ * MCUBoot uses continuous numbering for slot 0, slot 1, and the scratch
+ * while zephyr might numbering it differently.
```
Comment (utzig) [inline boot/zephyr/flash_map_extended.c]: "might **number** it differently"

**Comment by nvlsianpu** [comment]
Can be merged! Related Zephyr PRs were just merged to the zephyr master.

**Comment by nvlsianpu** [comment]
@mbolivar  ^^ Plasn to do it ASAP.

**Comment by nvlsianpu** [comment]
@carlescufi Can you approve?

**Pull request merged**